packages/fast-components-react-base/src/dialog/dialog.tsx

```diff
@@ -114,28 +114,49 @@ class Dialog extends Foundation<DialogHandledProps, DialogUnhandledProps, {}> {
             <div
                 className={classNames(this.props.managedClasses.dialog_modalOverlay)}
                 onClick={this.handleOverlayClick}
+                onTouchStart={this.handleOverlayTouch}
                 role={"presentation"}
                 tabIndex={-1}
+                style={{
+                    touchAction: "none",
+                }}
             />
         );
     }
 
+    /**
+     * Handle overlay touch
+     */
+    private handleOverlayTouch = (event: React.TouchEvent): void => {
+        this.checkForSoftDismiss(event);
+    };
+
+    /**
+     * Handle overlay click
+     */
     private handleOverlayClick = (event: React.MouseEvent): void => {
-        if (
-            this.props.onDismiss &&
-            typeof this.props.onDismiss === "function" &&
-            this.props.visible
-        ) {
-            this.props.onDismiss(event);
-        }
+        this.checkForSoftDismiss(event);
```

Comment:
To reduce falling out of sync by having to maintain two methods that do the same thing, can we make touch and click the same method? Perhaps `handleOverlaySelection`.